### PR TITLE
Call nvidia-smi.exe before running tests on Windows 

### PR DESCRIPTION
**Pull request by skyline75489**
Follow up for #53334

### CODE REVIEW

File: .jenkins/pytorch/win-test.sh [hunk 1]
```diff
@@ -45,6 +45,14 @@ if [[ "${CIRCLE_JOB}" == *11* ]]; then
...
 
 run_tests() {
+    if [[ "${CIRCLE_JOB}" == *10.1* ]]; then
+        # nvidia-smi for CUDA <= 10.1
+        "/c/Program Files/NVIDIA Corporation/NVSMI/nvidia-smi.exe"
+    elif [[ "${CIRCLE_JOB}" == *10.2* ]] || [[ "${CIRCLE_JOB}" == *11* ]]; then
+        # nvidia-smi for CUDA >= 10.2
+        "/c/Windows/System32/nvidia-smi.exe"
+    fi
+
```
Comment (malfet) [inline .jenkins/pytorch/win-test.sh:55]:
```suggestion
    # Run nvidia-smi if available
    for path in  /c/Program Files/NVIDIA Corporation/NVSMI/nvidia-smi.exe /c/Windows/System32/nvidia-smi.exe; do
    if [ -x $path ]; then
       $path;
       break
    fi
    done
```

**Review by malfet — CHANGES_REQUESTED** [review]
Hmm, existing check seems to rely on job names and some pre-defined location and will start to fail if location of NVIDIA-smi is moved.
How about iterating over potential paths and executing the program if it exists?
Something like
```
for p in /a /b;
  if [ -x $p/nvidia-smi.exe ];
     $p/nvidia-smi.exe
  fi
done
```

File: .jenkins/pytorch/win-test.sh (hunk 1 above)
Comment (skyline75489) [reply .jenkins/pytorch/win-test.sh]: Yeah thanks!

File: .jenkins/pytorch/win-test.sh [hunk 2]
```diff
@@ -45,6 +45,14 @@ if [[ "${CIRCLE_JOB}" == *11* ]]; then
 fi
 
 run_tests() {
+    # Run nvidia-smi if available
+    for path in  /c/Program Files/NVIDIA Corporation/NVSMI/nvidia-smi.exe /c/Windows/System32/nvidia-smi.exe; do
+        if [ -x $path ]; then
+            $path;
+             break
```
Comment (malfet) [inline .jenkins/pytorch/win-test.sh:52]: Nit
```suggestion
            $path;
            break
```

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**